Gems/Atom/RPI/Code/Include/Atom/RPI.Public/Pass/CopyPass.h

```diff
@@ -101,6 +109,36 @@ namespace AZ
             int m_currentBufferIndex = 0;
             AZStd::array<Ptr<RHI::Fence>, MaxFrames> m_device1SignalFence;
             AZStd::array<Ptr<RHI::Fence>, MaxFrames> m_device2WaitFence;
+            // RPI::Pass overrides...
+            TimestampResult GetTimestampResultInternal() const override;
+            PipelineStatisticsResult GetPipelineStatisticsResultInternal() const override;
+
+            // Helper function to get the query by the scope index and query type
+            RHI::Ptr<Query> GetQuery(ScopeQueryType queryType, int localDeviceIndex);
+
+            // Executes a lambda depending on the passed ScopeQuery types
+            template<typename Func>
+            void ExecuteOnTimestampQuery(Func&& func, int localDeviceIndex);
+            template<typename Func>
+            void ExecuteOnPipelineStatisticsQuery(Func&& func, int localDeviceIndex);
+
+            // Begin recording commands for the ScopeQueries
+            void BeginScopeQuery(const RHI::FrameGraphExecuteContext& context, int localDeviceIndex);
+
+            // End recording commands for the ScopeQueries
+            void EndScopeQuery(const RHI::FrameGraphExecuteContext& context, int localDeviceIndex);
```

Comment:
Why `localDeviceIndex` and not just `deviceIndex`?

Reply:
Since we typically use `deviceIndex` to denote an actual index of a device (i.e. the first GPU is GPU `0` etc.), but this index is just used to denote if the `sourceDevice` or the `destinationDevice` should be used (to index the local data structures), so it will always be 0 or 1.
Do you have a more fitting name?

Reply:
I think `scopeIndex` would make sense then.